pkg/virtual/framework/forwardingregistry/store.go

```diff
@@ -179,7 +181,13 @@ func DefaultDynamicDelegatedStoreFuncs(
 		doUpdate := func() (*unstructured.Unstructured, error) {
 			oldObj, err := s.Get(ctx, name, &metav1.GetOptions{})
 			if err != nil {
-				return nil, err
+				if !apierrors.IsNotFound(err) {
+					return nil, err
+				}
+				// This is required to make the call to UpdatedObjectInfo.UpdatedObject pass.
+				oldObj = &unstructured.Unstructured{
+					Object: make(map[string]interface{}),
+				}
```

Comment:
worth an e2e test?

Reply:
Definitely, working on it 👍🏼

Reply:
Added with 987273db6243279cb9ffa2b6d8f55dea57a0a78d.